### PR TITLE
pathToSpeaker simulating using limelight odometry and y button for control 

### PR DESCRIPTION
**Pull request by grace-mills**

### CODE REVIEW

File: Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java
Comment (jmullins16) [inline Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java]: Look at this API page at the "equals" method for Pose2d. Should be able to do something like:

if (rawPose != null || !rawPose.equals(origin))

https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Pose2d.html#%3Cinit%3E()

File: Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java
Comment (jmullins16) [inline Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java]: I think this .update call should already be called within the swerve library provided by CTRE (or else it would not have worked at all even when the LL wasn't there). So this might be applying the update a second time.

File: Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java
Comment (jmullins16) [inline Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java]: The pose is useful, but I'm not sure logging the individual swerve module positions is helpful. We should discuss.

File: Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java
Comment (jmullins16) [inline Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java]: When checking in the code, we need to leave this as "false" to prevent everyone else from having simulation problems. When we get working robots, we can leave it as "true". You'll need to set it to true when you are testing.

File: Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java
Comment (jmullins16) [inline Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:143]: Did we prove that we get a 0,0,0 pose with the valid setting as true? I don't remember the results.

File: Comp2024/src/main/java/frc/robot/subsystems/Vision.java
Comment (jmullins16) [inline Comp2024/src/main/java/frc/robot/subsystems/Vision.java]: Maybe these lines aren't needed anymore? It's commented out.

**Review by jmullins16 — CHANGES_REQUESTED** [review]
Looks pretty good. We should discuss my questions before making new changes and merging.

File: Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java
Comment (grace-mills) [reply Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:143]: Yes valid is a boolean in Limelight Helpers and we still had the 0,0,0 problem with it true

File: Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java
Comment (jmullins16) [inline Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:194]: This is a minor "nit", but origin is a local variable and should be "origin" (drinking camel case).

Comment (jmullins16) [reply Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:194]: I guess this is in Comp2023-imported anyway and not our Comp2024 codebase.

**Review by jmullins16 — APPROVED** [review]
Approving with only a minor request for capitalization of a variable. For the LL returning a zero pose, you're doing the right thing by keeping the check in the code for now and getting it merged. I believe that we'll learn a better way to detect the "valid" state in the future, because the LL has all of the information needed to mark a (0, 0, 0) pose as invalid--we've probably just not figured it out. But now is not the time to go down that rabbit hole. Nice work with all of the changes!

**Pull request merged**
